bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableDataSettingsFactory.java

```diff
@@ -0,0 +1,265 @@
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+
+import org.apache.hadoop.conf.Configuration;
+import org.threeten.bp.Duration;
+
+import com.google.api.gax.batching.BatchingSettings;
+import com.google.api.gax.batching.FlowControlSettings;
+import com.google.api.gax.core.CredentialsProvider;
+import com.google.api.gax.core.FixedCredentialsProvider;
+import com.google.api.gax.core.NoCredentialsProvider;
+import com.google.api.gax.grpc.GrpcTransportChannel;
+import com.google.api.gax.retrying.RetrySettings;
+import com.google.api.gax.rpc.FixedHeaderProvider;
+import com.google.api.gax.rpc.FixedTransportChannelProvider;
+import com.google.cloud.bigtable.config.BigtableOptions;
+import com.google.cloud.bigtable.config.BulkOptions;
+import com.google.cloud.bigtable.config.CredentialFactory;
+import com.google.cloud.bigtable.config.CredentialOptions;
+import com.google.cloud.bigtable.config.Logger;
+import com.google.cloud.bigtable.config.RetryOptions;
+import com.google.cloud.bigtable.data.v2.BigtableDataSettings;
+import com.google.cloud.bigtable.data.v2.BigtableDataSettings.Builder;
+import com.google.cloud.bigtable.data.v2.models.InstanceName;
+import com.google.cloud.bigtable.data.v2.stub.BigtableStubSettings;
+
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.internal.GrpcUtil;
+
+/**
+ * Static methods to convert an instance of {@link Configuration} or {@link BigtableOptions} to a
+ * {@link BigtableDataSettings} instance.
+ */
+public class BigtableDataSettingsFactory {
+  /** Constant <code>LOG</code> */
+  private static final Logger LOG = new Logger(BigtableDataSettingsFactory.class);
+
+  /**
+   * To create an instance of {@link BigtableDataSettings} from {@link BigtableOptions}.
+   *
+   * @param configuration a {@link BigtableOptions} object.
+   * @return a {@link BigtableDataSettings} object.
+   * @throws IOException if any.
+   */
+  public static BigtableDataSettings fromBigtableOptions(final BigtableOptions options)
+      throws IOException, GeneralSecurityException {
+    if (!options.getRetryOptions().enableRetries()) {
+      throw new IllegalStateException(
+          "Retry is must for BigtableDataSettings configuration from BigtableOptions.");
+    }
+
+    BigtableDataSettings.Builder builder = BigtableDataSettings.newBuilder();
+
+    InstanceName instanceName = InstanceName.newBuilder().setProject(options.getProjectId())
+        .setInstance(options.getInstanceId()).build();
+    builder.setInstanceName(instanceName);
+    builder.setAppProfileId(options.getAppProfileId());
+
+    LOG.debug("endpoint host %s.", options.getDataHost());
+    LOG.debug("endpoint host %s.", options.getPort());
+    builder.setEndpoint(options.getDataHost() + ":" + options.getPort());
+
+    buildCredentialProvider(builder, options.getCredentialOptions());
+
+    buildBulkOptions(builder, options);
```

Comment:
The naming is a bit inconsistent here, please name it after the method name like `buildBulkMutations`